website/api/controllers/docs/view-basic-documentation.js

```diff
@@ -44,15 +44,33 @@ module.exports = {
     // console.log('pageUrlSuffix:',pageUrlSuffix);
     // console.log('SECTION_URL_PREFIX + "/" + _.trim(pageUrlSuffix, "/"):',SECTION_URL_PREFIX + '/' + _.trim(pageUrlSuffix, '/'));
     // console.log('thisPage:',thisPage);
+
+    // if the url doesn't match any existing page, or the page it matches doesn't match the url provided
+    // then it might have extra slashes or uppercase characters (e.g. fleetdm.com/docs///usiNG-fleet////)
+    // Note: because this also handles the docs landing page and a `pageUrlSuffix` might not have
+    // been provided, we won't be rechecking the url if thisPage.url is '/docs'
+    if (!thisPage || (thisPage.url !== '/docs/'+pageUrlSuffix && thisPage.url !== '/docs')) {
+      // creating a regex to match instances of multiple slashes
+      let multipleSlashesRegex = /\/{2,}/g;
+      // Creating a lowercase double-slashless url to search with
+      let modifiedPageUrlSuffix = pageUrlSuffix.toLowerCase().replace(multipleSlashesRegex, '/');
+      // Finding the appropriate page content using the modified url.
+      let revisedPage = _.find(sails.config.builtStaticContent.markdownPages, {
+        url: _.trimRight(SECTION_URL_PREFIX + '/' + _.trim(modifiedPageUrlSuffix, '/'), '/')
+      });
+      // If we matched a page with the modified url, then redirect to that.
+      if(revisedPage && revisedPage.url) {
+        throw {redirect: revisedPage.url};
+      } else {
+        // otherwise, throw a 404 error.
+        throw 'notFound';
+      }
+    }
+
     if (!thisPage) {
       throw 'notFound';
```

Comment:
For readability (i.e. to make it more obvious when this code would be reached and when we return early instead), can you refactor to merge this with the above conditional that you added?

I think what I'd recommend for readability is first (1) deciding whether a redirect will be appropriate and getting that into a variable, then (2) using that variable as well as `!thisPage` to do your flow control in just a couple of lines of code  (a la what you've got going on in [lines 62-67](https://github.com/fleetdm/fleet/pull/2083/files#diff-126576e032f33963a5fc0d7598256a737605367560db8d43315d64da5e78ec03R62-R67) in your additions, but outside of any nesting)